### PR TITLE
New Feature : Web Plant CRUD DONE

### PR DESCRIPTION
**Pull request by mnaufalrifqir**

### CODE REVIEW

**Review by widhofaisal — APPROVED** [review]
mantap mas 👍👍⚡⚡

File: controller/admin/plant_page_controller.go [hunk 1]
```diff
@@ -0,0 +1,585 @@
...
+
+	// Get all plants
+	if err := config.DB.Find(&plants).Error; err != nil {
+		log.Print(color.RedString(err.Error()))
+		return c.JSON(http.StatusBadRequest, map[string]interface{}{
+			"status":  400,
+			"message": "bad request",
+		})
+	}
+	//Populate all field for each plant to response
```
Comment (dvnggak) [inline controller/admin/plant_page_controller.go:71]: LGTM !! 👍🏻

File: controller/admin/plant_page_controller.go [hunk 2]
```diff
@@ -0,0 +1,585 @@
...
+
+	plants := []model.Plant{}
+	responses := []Response{}
+
+	// Get all plants
+	if err := config.DB.Where("`name` LIKE ?", "%"+keyword+"%").Find(&plants).Error; err != nil {
+		log.Print(color.RedString(err.Error()))
+		return c.JSON(http.StatusBadRequest, map[string]interface{}{
+			"status":  400,
+			"message": "bad request",
```
Comment (dvnggak) [inline controller/admin/plant_page_controller.go:123]: nice consistency of the error message 💯

File: model/admin.go [hunk 3]
```diff
@@ -52,12 +53,84 @@ type Weather struct {
...
+
 type Picture struct {
-	gorm.Model
-	URL       string `json:"url"`
-	ArticleID *uint  `json:"article_id" `
-	ProductID *uint  `json:"product_id"`
-	WeatherID *uint  `json:"weather_id"`
+	gorm.Model        `json:"-"`
+	URL               string `json:"url"`
+	ArticleID         *uint  `json:"-"`
```
Comment (dvnggak) [inline model/admin.go:125]: LGTM 👍🏻

File: utils/jwt.go [hunk 4]
```diff
@@ -72,3 +72,29 @@ func GetUserIDFromToken(tokenString string) (uint, error) {
...
+		return []byte(constant.SECRET_JWT), nil
+	})
+
+	if err != nil {
+		return 0, err
+	}
+
+	// Check if the token is valid
+	if claims, ok := token.Claims.(jwt.MapClaims); ok && token.Valid {
+		// Extract the user_id claim
```
Comment (dvnggak) [inline utils/jwt.go:93]: Neat solution nice 🥇

**Review by dvnggak — APPROVED** [review]

**Pull request merged**